blockprod/src/detail/mod.rs

```diff
@@ -260,6 +260,16 @@ impl BlockProduction {
         transactions_source: TransactionsSource,
         custom_id: Option<Vec<u8>>,
     ) -> Result<(Block, oneshot::Receiver<usize>), BlockProductionError> {
+        let is_initial_block_download = self
+            .chainstate_handle
+            .call(|this| this.info().map(|info| info.is_initial_block_download))
+            .await?
+            .map_err(|_| BlockProductionError::ChainstateInfoRetrievalError)?;
+
+        if is_initial_block_download {
+            return Err(BlockProductionError::ChainstateWaitForSync);
+        }
+
```

Comment:
I think this change now puts us in a Catch-22 - blockprod tests fail because we start out at Genesis, but we can't move forward because of line 269.

I've been looking at possible solutions, but they're all hacks which I don't want to do (plus they won't work for functional tests).

Any suggestions?

Reply:
Thinking the further, I guess we can gate that `if` above with a `BlockProdConfig` param, then bypass it within unit and functional tests?

Reply:
Having a similar issue in mempool, which is also supposed to be disabled during IBD. Tried to fiddle with the `MaxTipAge` setting in tests but if the chain is at genesis, it is considered in the IBD stage no matter what. Maybe we could make it respect max tip age even at genesis?